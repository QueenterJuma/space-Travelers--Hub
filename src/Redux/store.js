import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './Missions/missionSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
  },
});

export default store;
