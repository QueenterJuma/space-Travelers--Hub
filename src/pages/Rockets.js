import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../Redux/rockets/rocketsSlice';
import Rocket from '../Components/Rocket';

const Rockets = () => {
  const dispatch = useDispatch();
  const { rockets } = useSelector((store) => store.rockets);
  useEffect(() => {
    if (!rockets.length) {
      dispatch(fetchRockets());
    }
  }, [dispatch]);

  return (
    <div className="rockets">
      {rockets.map((rocket) => (
        <Rocket key={rocket.id} rocket={rocket} />
      ))}
    </div>
  );
};

export default Rockets;
