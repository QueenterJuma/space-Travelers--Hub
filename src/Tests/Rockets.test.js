import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../Redux/store';
import Rockets from '../pages/Rockets';

describe('Test Rockets Component', () => {
  it('should render Rockets component', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
