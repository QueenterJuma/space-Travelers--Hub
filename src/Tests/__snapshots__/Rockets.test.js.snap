// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rockets Component should render Rockets component 1`] = `
<div
  className="rockets"
/>
`;
